tests/test_console.py

```diff
@@ -81,18 +84,22 @@ def submit_test_responses():
             db.session.commit()
 
 
+@testing.postgresql.skipIfNotInstalled
 class TestAuthentication(unittest.TestCase):
 
     def setUp(self):
-        self.postgres = Postgresql()
-        Postgresql.clear_cache()
         self.app = server.app.test_client()
         self.app.testing = True
         self.render_templates = False
 
+        with app.app_context():
+            db.create_all()
+            create_initial_users()
+            db.session.commit()
+
     def tearDown(self):
         Postgresql.clear_cache()
-        self.postgres.stop()
+        postgresql.stop()
```

Comment:
this should probably still reference self.postgresql?

Reply:
The reason I didn't do this was that if the postgresql object is generated as a member of the test classes the database URI changes. It seemed cleaner to generate it outside the classes. If you think it's better practice I will move it back inside the test classes.

Reply:
I've gone over it again and realised you were correct, it was not behaving as intended.